templates/.semaphore/semaphore.yml

```diff
@@ -0,0 +1,55 @@
+version: "v1.0"
+name: Pipeline with parallel blocks and dependencies
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+auto_cancel:
+  running:
+    when: "true"
+blocks:
+  - name: Install dependencies
+    dependencies: []
+    task:
+      secrets:
+        - name: [project-name]
+      jobs:
+        - name: cache
+          commands:
+            - checkout
+            - cache restore bundle-$SEMAPHORE_GIT_BRANCH-$(checksum Gemfile.lock),bundle-$SEMAPHORE_GIT_BRANCH,bundle-master
```

Comment:
this is done by simply writing `cache restore`

Reply:
But only if the gems are installed to inside the working directory. This has to be done like this on line 21:

    bundle install --deployment -j 4 --path vendor/bundle